apps/ewallet/lib/ewallet/web/url_validator.ex

```diff
@@ -4,13 +4,29 @@ defmodule EWallet.Web.UrlValidator do
   """
 
   @doc """
-  Checks that the given url is allowed as a redirect url.
+  Checks that the given url is allowed as a redirect url in the application settings.
   """
   def allowed_redirect_url?(url) do
     base_url = Application.get_env(:ewallet, :base_url)
     prefixes = Application.get_env(:ewallet, :redirect_url_prefixes)
     allowed = [base_url | prefixes]
 
-    Enum.any?(allowed, fn prefix -> String.starts_with?(url, prefix) end)
+    Enum.any?(allowed, fn prefix -> allowed_redirect_url?(url, prefix) end)
+  end
+
+  @doc """
+  Checks that the given url is allowed by the given prefix
+  """
+  def allowed_redirect_url?(url, prefix) do
+    # Add trailing slashes to prevent urls such as 'https://example.comnotexample.com'
+    # matching 'https://example.com'
+    url = trailing_slashed(url)
+    prefix = trailing_slashed(prefix)
```

Comment:
Use `URI.merge`:

```
iex(5)> "https://foo.bar" |> URI.merge("/foo/bar/baz") |> URI.to_string()
"https://foo.bar/foo/bar/baz"
```

Reply:
Since `URI.merge("https://foo.bar/some/segment", "/")` would return `"https://foo.bar/"` losing the URI segments, I've reverted back to basic string matching instead.